apisix/init.lua

```diff
@@ -629,6 +630,12 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
+    local status_code = tonumber(get_var("upstream_status"))
```

Comment:
Note that `upstream_status` could be `500, 502` or other similar stuff if retry between upstreams is involved.

Reply:
How to deal with this situation, do you have any good suggestions?

Reply:
What about checking the last status ("5xx", or " 5xx")?

Reply:
Yes this is a good way.

Reply:
This is still unresolved.

Reply:
If the upstream retry, the status code returns 500, 502 or other similar status codes. This situation is not returned by apisix itself, this should be the status code of the upstream service. We can capture it with `get_var("status")`. If it is the status code returned by apisix itself, the value captured by `get_var("status")` will be a nil.
This is my opinion. At present, I have not thought of a better solution.

Reply:
`get_var("upstream_status")`, is this wrong? 
please confirm do we need to fix it?

Reply:
I think there is a problem. When it comes to retries between upstreams and all upstream services are unavailable, the value of `$upstream_status` is nil. APISIX itself returns a 502 status code. I think this 502 status code should be the status code of the upstream service.
This is the relevant test case：
https://github.com/apache/apisix/pull/2817/files#diff-8b1dc592c5a25c17fc62106ed68fd5b910eb584064cba052b1af9088d3e74ff5R286-R326

Reply:
It is because `tonumber("500, 502")` will be `nil`.

Reply:
I understand, thank you very much. I will update later.